Plugins/Skinning/src/SkinningComponent.cpp

```diff
@@ -320,72 +314,73 @@ void SkinningComponent::endSkinning() {
-        m_frameData.m_currentPose   = m_refData.m_refPose;
-        m_frameData.m_previousPose  = m_refData.m_refPose;
-        m_frameData.m_currentPos    = m_refData.m_referenceMesh.vertices();
-        m_frameData.m_previousPos   = m_refData.m_referenceMesh.vertices();
-        m_frameData.m_currentNormal = m_refData.m_referenceMesh.normals();
-        m_frameData.m_refToCurrentRelPose =
-            Ra::Core::Animation::relativePose( m_frameData.m_currentPose, m_refData.m_refPose );
-        m_frameData.m_prevToCurrentRelPose = Ra::Core::Animation::relativePose(
-            m_frameData.m_currentPose, m_frameData.m_previousPose );
-    }
 }
 
-void SkinningComponent::handleWeightsLoading( const Ra::Core::Asset::HandleData* data,
-                                              const std::string& meshName ) {
+void SkinningComponent::handleSkinDataLoading( const Ra::Core::Asset::HandleData* data,
+                                               const std::string& meshName,
+                                               const Ra::Core::Transform& meshFrame ) {
     m_contentsName = data->getName();
     m_meshName     = meshName;
     setupIO( meshName );
+    m_meshFrameInv = meshFrame.inverse();
     for ( const auto& bone : data->getComponentData() )
     {
-        auto it = bone.m_weight.find( meshName );
-        if ( it != bone.m_weight.end() ) { m_loadedWeights[bone.m_name] = it->second; }
+        auto it_w = bone.m_weights.find( meshName );
+        if ( it_w != bone.m_weights.end() )
+        {
+            m_loadedWeights[bone.m_name] = it_w->second;
+            auto it_b                    = bone.m_bindMatrices.find( meshName );
+            if ( it_b != bone.m_bindMatrices.end() )
+            { m_loadedBindMatrices[bone.m_name] = it_b->second; }
+            else
+            {
+                LOG( logWARNING ) << "Bone " << bone.m_name
+                                  << " has skinning weights but no bind matrix. Using Identity.";
+                m_loadedBindMatrices[bone.m_name] = Ra::Core::Transform::Identity();
+            }
+        }
     }
 }
 
 void SkinningComponent::createWeightMatrix() {
-    m_refData.m_weights.resize( m_refData.m_referenceMesh.vertices().size(),
+    m_refData.m_weights.resize( int( m_refData.m_referenceMesh.vertices().size() ),
                                 m_refData.m_skeleton.size() );
-    for ( int col = 0; col < m_refData.m_skeleton.size(); ++col )
+    std::vector<Eigen::Triplet<Scalar>> triplets;
+    for ( uint col = 0; col < m_refData.m_skeleton.size(); ++col )
     {
-        auto it = m_loadedWeights.find( m_refData.m_skeleton.getLabel( col ) );
+        std::string boneName = m_refData.m_skeleton.getLabel( col );
+        auto it              = m_loadedWeights.find( boneName );
         if ( it != m_loadedWeights.end() )
         {
             const auto& W   = it->second;
-            const uint size = W.size();
+            const uint size = uint( W.size() );
```

Comment:
why storing `size` ?
Why not simply `for ( uint i = 0; i < uint( W.size() ); ++i) ... ` ?